internal/controller/watchfilters/watch_filters.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2025, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package watchfilters
```

Comment:
I feel like this package does not belong here, as all others are controller packages. This could indicate there is a kind called `WatchFilters`. Shall we think of a different place? Possibly inside the internal package.

Reply:
can't we move to pkg?

Reply:
Shall we keep it inside internal package as we discussed?

Reply:
fixed